src/meta/src/hummock/compaction/compaction_picker.rs

```diff
@@ -64,7 +64,7 @@ impl CompactionPicker for MinOverlappingPicker {
             let mut pending_campct = false;
             let overlap_files = self
                 .overlap_strategy
-                .check_base_level_overlap(&[table.clone()], &levels[target_level].table_infos);
+                .check_overlap_with_tables(&[table.clone()], &levels[target_level].table_infos);
```

Comment:
why changed with `check_overlap_with_tables` ?
`check_base_level_overlap` can use binary search to avoid compare every table in target level.

Reply:
If you want to use MinOverlappingPicker for hash-partition compaction, you can replace `overlap_strategy` with other algorithm.

Reply:
> If you want to use MinOverlappingPicker for hash-partition compaction, you can replace `overlap_strategy` with other algorithm.

L1 would have inevitable overlap key range after hash compaction

Reply:
I get. I just said we can replace it with a new kind of `OverlapStrategy`.

```rust
struct HashOverlapStrategy {
}

impl OverlapStrategy for HashOverlapStrategy {
}

```

Reply:
Because we can set different configure for every HummockGroup (Maybe different MView), It's not necessary to backup to `check_overlap_with_tables`.

Reply:
> Because we can set different configure for every HummockGroup (Maybe different MView), It's not necessary to backup to `check_overlap_with_tables`.

thank you. fixed.

Reply:
> The `HashOverlapStrategy` is OK. And this line of code can be revert.

sorry, I forgot to revert this